Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift

```diff
@@ -788,19 +785,78 @@ public class ParseDiagnosticsGenerator: SyntaxAnyVisitor {
         handledNodes: [name.id]
       )
     }
-    if let unexpected = node.unexpectedBetweenDeinitKeywordAndBody,
-      let signature = unexpected.compactMap({ $0.as(FunctionSignatureSyntax.self) }).only
+    if let unexpected = node.unexpectedBetweenDeinitKeywordAndEffectSpecifiers,
+      let params = unexpected.compactMap({ $0.as(ParameterClauseSyntax.self) }).only
     {
       addDiagnostic(
-        signature,
+        params,
         .deinitCannotHaveParameters,
         fixIts: [
-          FixIt(message: RemoveNodesFixIt(signature), changes: .makeMissing(signature))
+          FixIt(message: RemoveNodesFixIt(params), changes: .makeMissing(params))
+        ],
+        handledNodes: [params.id]
+      )
+    }
+
+    return .visitChildren
+  }
+
+  public override func visit(_ node: DeinitEffectSpecifiersSyntax) -> SyntaxVisitorContinueKind {
+    let throwsTokens: [TokenKind] = [
+      .keyword(.throws),
+      .keyword(.rethrows),
+      .keyword(.try),
+      .keyword(.throw),
+    ]
```

Comment:
Can’t you use the `ThrowsEffectSpecifier` token spec set here instead of re-defining the list?

Reply:
Done